osf/management/commands/migrate_pagecounter_data.py

```diff
@@ -0,0 +1,116 @@
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import connection
+
+from framework import sentry
+
+logger = logging.getLogger(__name__)
+
+
+LIMIT_CLAUSE = ' LIMIT %s);'
+NO_LIMIT_CLAUSE = ');'
+
+REVERSE_SQL_BASE = '''
+UPDATE osf_pagecounter PC
+SET
+    resource_id = NULL,
+    file_id = NULL,
+    version = NULL
+WHERE PC.id IN (
+    SELECT PC.id FROM osf_pagecounter PC
+        INNER JOIN osf_guid Guid on Guid._id = split_part(PC._id, ':', 2)
+        INNER JOIN osf_basefilenode File on File._id = split_part(PC._id, ':', 3)
+'''
+REVERSE_SQL = '{} {}'.format(REVERSE_SQL_BASE, NO_LIMIT_CLAUSE)
+REVERSE_SQL_LIMITED = '{} {}'.format(REVERSE_SQL_BASE, LIMIT_CLAUSE)
+
+FORWARD_SQL_BASE = '''
+    UPDATE osf_pagecounter PC
+    SET
+        action = split_part(PC._id, ':', 1),
+        resource_id = Guid.id,
+        file_id = File.id,
+        version = NULLIF(split_part(PC._id, ':', 4), '')::int
+    FROM osf_guid Guid, osf_basefilenode File
+        WHERE
+              Guid._id = split_part(PC._id, ':', 2) AND
+              File._id = split_part(PC._id, ':', 3) AND
+              PC.id in (
+                  select PC.id from osf_pagecounter PC
+                      inner join osf_guid Guid on Guid._id = split_part(PC._id, ':', 2)
+                      inner join osf_basefilenode File on File._id = split_part(PC._id, ':', 3)
+                  where (PC.resource_id is NULL or PC.file_id IS NULL) AND
+                        Guid._id IS NOT NULL AND File._id IS NOT NULL
```

Comment:
With the inner joins you don't need 
` Guid._id IS NOT NULL AND File._id IS NOT NULL`